File: src/main/java/com/codingforcookies/betterrecords/api/record/IRecord.java
package com.codingforcookies.betterrecords.api.record;

import com.codingforcookies.betterrecords.api.wire.IRecordWireHome;
import net.minecraft.item.ItemStack;

public interface IRecord {
    boolean isRecordValid(ItemStack itemStack);
    void onRecordInserted(IRecordWireHome wireHome, ItemStack itemStack);
}

File: src/main/kotlin/com/codingforcookies/betterrecords/api/record/IRecord.kt
package com.codingforcookies.betterrecords.api.record

import com.codingforcookies.betterrecords.api.wire.IRecordWireHome
import net.minecraft.item.ItemStack

interface IRecord {

    fun isRecordValid(itemStack: ItemStack): Boolean

    fun onRecordInserted(wireHome: IRecordWireHome, itemStack: ItemStack)
}
